esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -382,12 +404,25 @@ def __init__(
                 )
         self.resolution = resolution
 
+        if src_mask is True:
+            src_mask = _get_mask(src_mesh_cube)
+        elif src_mask is False:
+            src_mask = None
+        self.src_mask = src_mask
+        if tgt_mask is True:
+            tgt_mask = _get_mask(target_grid_cube)
+        elif tgt_mask is False:
+            tgt_mask = None
+        self.tgt_mask = tgt_mask
```

Comment:
I'd like to argue for keeping things mostly as they are. I think there is a bit of confusion here that could probably be cleared up by renaming variables. I'm thinking of having the top level, which could contain a boolean which switches behaviour or an array, could be called `use_src_mask` or `use_tgt_mask`. Once the logic around `_get_mask` has been called, I'm thinking of having this variable be called `src_mask` or `tgt_mask`. This variable is either `None` or an array and is what is passed to a lower level. 

There is perhaps an argument for `src_mask` usingg `False` rather than `None`, but I feel like `None` is a better reflection of what is happening mechanically.

I would very much argue against `None` being used in `use_src_mask` to mean you should use the source mask. I could imagine there being the expectation that if a regridder has the attribute `src_mask=None` then setting up `use_src_mask=None` would give you this outcome. This is actually the current behaviour when initialising a regridder, though it is undocumented since it is somewhat of a side effect and it behaves slightly differently in `ESMFAreaWeighted.regrid` where `None` is used to mean the value used in the initialisation of `ESMFAreaWeighted` should be used.

Reply:
See https://github.com/SciTools-incubator/iris-esmf-regrid/pull/219/commits/ed880838184511b84efd63072f137ff25c4e0933